qiskit/pulse/schedule.py

```diff
@@ -37,11 +44,14 @@ def __init__(self, *schedules: List[Union[ScheduleComponent, Tuple[int, Schedule
         Args:
             *schedules: Child Schedules of this parent Schedule. May either be passed as
                 the list of schedules, or a list of (start_time, schedule) pairs
-            name: Name of this schedule
+            name: Name of this schedule. Defaults to an autogenerated string if not provided.
 
         Raises:
             PulseError: If timeslots intercept.
         """
+        if name is None:
```

Comment:
As schedules could be parallelized we may run into similar issues as here #3491. Perhaps this PID extraction should be made into a utility function and then used here as well?

Reply:
oh woops! Forgot to ask about that, thanks for catching. I guess it would eventually belong in the common naming mechanism, but for now should we put it under `qiskit.util`?

Reply:
I think that is fine. Once you do that we can ask those who are more familiar with that code path if that is the appropriate location.